web/packages/teleport/src/Account/ManageDevices/AddAuthDeviceWizard/AddAuthDeviceWizard.tsx

```diff
@@ -246,51 +284,160 @@ function CreatePasskeyStep({
   );
 }
 
+function CreateMfaBox({
+  auth2faType,
+  newMfaDeviceType,
+  privilegeToken,
+  onNewMfaDeviceTypeChange,
+}: {
+  auth2faType: Auth2faType;
+  newMfaDeviceType: Auth2faType;
+  privilegeToken: string;
+  onNewMfaDeviceTypeChange(o: Auth2faType): void;
+}) {
+  const mfaOptions = createMfaOptions({
+    auth2faType,
+    required: true,
+  }).map((o: MfaOption) =>
+    // Be more specific about the WebAuthn device type (it's not a passkey).
+    o.value === 'webauthn' ? { ...o, label: 'Hardware Device' } : o
+  );
+
+  return (
+    <>
+      Multi-factor type
+      <RadioGroup
+        name="mfaOption"
+        options={mfaOptions}
+        value={newMfaDeviceType}
+        flexDirection="row"
+        gap={3}
+        onChange={o => {
+          onNewMfaDeviceTypeChange(o as Auth2faType);
+        }}
+      />
+      {newMfaDeviceType === 'otp' && (
+        <QrCodeBox privilegeToken={privilegeToken} />
+      )}
+    </>
+  );
+}
+
+function QrCodeBox({ privilegeToken }: { privilegeToken: string }) {
+  const fetchQrCodeAttempt = useAttempt('');
+  const [qrCode, setQrCode] = useState('');
+
+  useEffect(() => {
+    fetchQrCodeAttempt.run(() =>
+      auth
+        .createMfaRegistrationChallenge(privilegeToken, 'totp')
+        .then(res => setQrCode(res.qrCode))
+    );
+  }, []);
+
+  return (
+    <Flex
+      flexDirection="column"
+      borderRadius={8}
+      p={3}
+      mb={4}
+      css={`
+        background: ${props => props.theme.colors.interactive.tonal.neutral[0]};
+      `}
+    >
+      <Flex height="168px" justifyContent="center" alignItems="center">
+        {fetchQrCodeAttempt.attempt.status === 'processing' && <Indicator />}
+        {fetchQrCodeAttempt.attempt.status === 'success' && (
+          <Image
+            src={`data:image/png;base64,${qrCode}`}
+            height="100%"
+            style={{
+              boxSizing: 'border-box',
+              border: '8px solid white',
+            }}
+          />
+        )}
+      </Flex>
+      <Text typography="body1" textAlign="center" mt={2}>
+        Scan the QR Code with any authenticator app.
+        <br />
+        We recommend{' '}
+        <Link href="https://authy.com/download/" target="_blank">
+          Authy
```

Comment:
I know this has always been here, but I wonder if it should stay.

Do we actually recommend authy? If so, why? I don't know that anyone can answer these questions.

cc @xinding33 @roraback what do you guys think about this?